File: IntToRoman/IntToRoman.ts
const IntToRoman = (num: number): string => {
  const symbolMap = [
    { value: 1000, keys: "M" },
    { value: 900, keys: "CM" },
    { value: 500, keys: "D" },
    { value: 400, keys: "CD" },
    { value: 100, keys: "C" },
    { value: 90, keys: "XC" },
    { value: 50, keys: "L" },
    { value: 40, keys: "XL" },
    { value: 10, keys: "X" },
    { value: 9, keys: "IX" },
    { value: 5, keys: "V" },
    { value: 4, keys: "IV" },
    { value: 1, keys: "I" },
  ];
  
  let result = "" as string;
  for (const { value, keys } of symbolMap) {
    const count = Math.floor(num / value);
    result += keys.repeat(count);
    num -= value * count;
  }

  return result;
};

console.log(IntToRoman(1244));

File: Java/Algorithms/src/algorithms/IntToRoman.java
package algorithms;

import java.util.LinkedHashMap;
import java.util.Map;

public class IntToRoman {
    public String intToRoman(int num) {
        Map<Integer, String> symbolMap = new LinkedHashMap<>();
        symbolMap.put(1000, "M");
        symbolMap.put(900, "CM");
        symbolMap.put(500, "D");
        symbolMap.put(400, "CD");
        symbolMap.put(100, "C");
        symbolMap.put(90, "XC");
        symbolMap.put(50, "L");
        symbolMap.put(40, "XL");
        symbolMap.put(10, "X");
        symbolMap.put(9, "IX");
        symbolMap.put(5, "V");
        symbolMap.put(4, "IV");
        symbolMap.put(1, "I");

        StringBuilder result = new StringBuilder();

        for (Map.Entry<Integer, String> entry : symbolMap.entrySet()) {

            int key = entry.getKey();
            String value = entry.getValue();

            int count = num / key;
            if (count > 0) {
                result.append(value.repeat(count));
                num -= key * count;
            }
        }
        return result.toString();
    }
}
